codegen/src/main/resources/software/amazon/awssdk/codegen/rules2/RulesFunctions.java.resource

```diff
@@ -6,49 +6,43 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Supplier;
-import java.util.regex.Pattern;
 import software.amazon.awssdk.annotations.SdkInternalApi;
 import software.amazon.awssdk.core.exception.SdkClientException;
 
 @SdkInternalApi
 public class RulesFunctions {
-    private static final Pattern VALID_HOST_LABEL_SUBDOMAINS = Pattern.compile("[a-zA-Z\\d][a-zA-Z\\d\\-.]{0,62}");
-    private static final Pattern VALID_HOST_LABEL = Pattern.compile("[a-zA-Z\\d][a-zA-Z\\d\\-]{0,62}");
+    private static final String[] ENCODED_CHARACTERS = { "+", "*", "%7E" };
+    private static final String[] ENCODED_CHARACTERS_REPLACEMENTS = { "%20", "%2A", "~" };
 
-    private static final Pattern VIRTUAL_HOSTABLE_BUCKET = Pattern.compile("[a-z\\d][a-z\\d\\-.]{1,61}[a-z\\d]");
-    private static final Pattern VIRTUAL_HOSTABLE_BUCKET_NO_SUBDOMAINS = Pattern.compile("[a-z\\d][a-z\\d\\-]{1,61}[a-z\\d]");
-    private static final Pattern NO_IPS = Pattern.compile("(\\d+\\.){3}\\d+");
-    private static final Pattern NO_CONSECUTIVE_DASH_OR_DOTS = Pattern.compile(".*[.-]{2}.*");
+    private static final LazyValue<PartitionData> PARTITION_DATA = LazyValue.<PartitionData> builder()
+            .initializer(RulesFunctions::loadPartitionData).build();
 
-    private static final String[] ENCODED_CHARACTERS = {"+", "*", "%7E"};
-    private static final String[] ENCODED_CHARACTERS_REPLACEMENTS = {"%20", "%2A", "~"};
+    private static final LazyValue<Partition> AWS_PARTITION = LazyValue.<Partition> builder()
```

Comment:
Do we have existing test case that will test this patterns and make sure they we following the pattern ?